src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -177,6 +179,36 @@ public function getEnv($prefix, $name, \Closure $getEnv)
             return $env;
         }
 
+        if ('url' === $prefix) {
+            $env = \parse_url($env);
+
+            if (false === $env) {
+                throw new RuntimeException(sprintf('Invalid URL in env var "%s"', $name));
+            }
+
+            $env += array(
+                'scheme' => '',
+                'host' => '',
+                'port' => 0,
```

Comment:
Now `default` processor is merged, I think we can remove those default values. What do you think?

```
port: '%env(default:default_port:key:port:url:DNS)%`
```

User'll have to create many default parameters :-/